multi-node/aws/pkg/cluster/cluster_test.go

```diff
@@ -11,16 +11,44 @@ import (
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/route53"
 	"github.com/coreos/coreos-kubernetes/multi-node/aws/pkg/config"
+	yaml "gopkg.in/yaml.v2"
 )
 
-const minimalConfigYaml = `
+/*
+TODO(colhom): when we fully deprecate instanceCIDR/availabilityZone, this block of
+logic will go away and be replaced by a single constant string
+*/
+func defaultConfigValues(t *testing.T, configYaml string) string {
+	defaultYaml := `
 externalDNSName: test.staging.core-os.net
 keyName: test-key-name
 region: us-west-1
-availabilityZone: us-west-1c
 clusterName: test-cluster-name
 kmsKeyArn: "arn:aws:kms:us-west-1:xxxxxxxxx:key/xxxxxxxxxxxxxxxxxxx"
 `
+	yamlStr := defaultYaml + configYaml
+
+	c := config.Cluster{}
+	if err := yaml.Unmarshal([]byte(yamlStr), &c); err != nil {
+		t.Errorf("failed umarshalling config yaml: %v :\n%s", err, yamlStr)
+	}
+
+	if len(c.Subnets) > 0 {
+		for i := range c.Subnets {
+			c.Subnets[i].AvailabilityZone = fmt.Sprintf("dummy-az-%d", i)
+		}
+	} else {
+		//Legacy behavior
+		c.AvailabilityZone = "dummy-az-0"
+	}
+
+	out, err := yaml.Marshal(&c)
```

Comment:
This is why 5341bf33e13e589399444846506d431c5894e512 exists. We're using the `config.Cluster` to automate pre-processing of test yaml data.